test/e2e/specs/site-editor/browser-history.spec.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+const baseUrl = process.env.WP_BASE_URL || 'http://localhost:8889';
+
+test.describe( 'Site editor browser history', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'Back button works properly', async ( { admin, page } ) => {
+		// Navigate to a template.
+		await admin.visitAdminPage( 'index.php' );
+		await admin.visitSiteEditor();
+
+		expect( page.url() ).toEqual( baseUrl + '/wp-admin/site-editor.php' );
```

Comment:
You can use `toHaveURL` and omit `baseUrl`:

```suggestion
		await expect( page ).toHaveURL( '/wp-admin/site-editor.php' );
```